FileCopy/utils.cpp

```
@@ -1,13 +1,52 @@
 #include "utils.h"
 
+#include <sys/stat.h>
+
+#include <cassert>
+#include <cstdlib>
 #include <fstream>
 
 #include "c150debug.h"
+#include "c150nastyfile.h"
 
 using namespace C150NETWORK;
 using namespace std;
 
-void setUpDebugLogging(const char *logname, int argc, char *argv[]) {
+bool check_directory(const char *dirname) {
+    assert(dirname);
+    struct stat statbuf;
+    if (lstat(dirname, &statbuf) != 0) {
+        fprintf(stderr, "Error stating supplied source directory %s\n",
+                dirname);
+        exit(8);
+    }
+
+    if (!S_ISDIR(statbuf.st_mode)) {
+        fprintf(stderr, "File %s exists but is not a directory\n", dirname);
+        exit(8);
+    }
+    return false;
+}
```

If we're going for lower line count we could combine the conditionals and leave the error message something like `"check_directory: %s is not a directory"`

---

```assert(dirname);
     struct stat statbuf;
     if (lstat(dirname, &statbuf) != 0 || !S_ISDIR(statbuf.st_mode)) {
         fprintf(stderr, "check_directory: %s is not a directory\n",
                 dirname);
         exit(8);
     }
```

---

I also think this should throw a runtime exception instead of printing and exiting... Since this function is essentially an assertion

---

Absolutely!